RecoLocalCalo/HcalRecAlgos/interface/timeshift_ns_hbheho.h

```diff
@@ -0,0 +1,14 @@
+#ifndef RecoLocalCalo_HcalRecAlgos_timeshift_ns_hbheho_h_
+#define RecoLocalCalo_HcalRecAlgos_timeshift_ns_hbheho_h_
+
+///Timeshift correction for HPDs based on the position of the peak ADC measurement.                                            
+///  Allows for an accurate determination of the relative phase of the pulse shape from                                        
+///  the HPD.  Calculated based on a weighted sum of the -1,0,+1 samples relative to the peak                                  
+///  as follows:  wpksamp = (0*sample[0] + 1*sample[1] + 2*sample[2]) / (sample[0] + sample[1] + sample[2])                    
+///  where sample[1] is the maximum ADC sample value.                                                                          
+float timeshift_ns_hbheho(float wpksamp);
+
+/// Special energy correction for some HB- cells
+float hbminus_special_ecorr(int ieta, int iphi, double energy, int runnum);
```

Comment:
Fine. Renamed into HcalCorrectionFunctions.h

On 06/26/2016 12:38 PM, Slava Krutelyov wrote:
>
> In RecoLocalCalo/HcalRecAlgos/interface/timeshift_ns_hbheho.h <https://github.com/cms-sw/cmssw/pull/14957#discussion_r68506304>:
>
> > @@ -0,0 +1,14 @@
> > +#ifndef RecoLocalCalo_HcalRecAlgos_timeshift_ns_hbheho_h_
> > +#define RecoLocalCalo_HcalRecAlgos_timeshift_ns_hbheho_h_
> > +
> > +///Timeshift correction for HPDs based on the position of the peak ADC measurement.
> > +///  Allows for an accurate determination of the relative phase of the pulse shape from
> > +///  the HPD.  Calculated based on a weighted sum of the -1,0,+1 samples relative to the peak
> > +///  as follows:  wpksamp = (0*sample[0] + 1*sample[1] + 2*sample[2]) / (sample[0] + sample[1] + sample[2])
> > +///  where sample[1] is the maximum ADC sample value.
> > +float timeshift_ns_hbheho(float wpksamp);
> > +
> > +/// Special energy correction for some HB- cells
> > +float hbminus_special_ecorr(int ieta, int iphi, double energy, int runnum);
>
> a file named timeshift containing energy correction is not intuitive.
> CorrectionFunctions.h may be well covering what's here.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/14957/files/0f0937650a7012d4c9a9a387e9ebb0997eb4842c#r68506304>, or mute the thread 
> <https://github.com/notifications/unsubscribe/AFS4-QiZlp5TynSgw4i4oQb5CLYCC_g-ks5qPrj-gaJpZM4I8hdG>.
>